src/core/qgstiles.cpp

```diff
@@ -35,6 +38,33 @@ QgsTileMatrix QgsTileMatrix::fromWebMercator( int zoomLevel )
   return tm;
 }
 
+QgsTileMatrix QgsTileMatrix::fromCustomDef( int zoomLevel, const QgsCoordinateReferenceSystem &crs, const QgsPointXY &z0TopLeftPoint, const double &scaleDenom0 )
```

Comment:
In my opinion, we should be defining tile zero by CRS + top-left XY + length of the side of square (in map units) ... instead of using CRS + top-left XY + scale.

To me, scale at the level of the whole world is not very well defined - it varies greatly in different parts of the root tile, while the dimension of the tile square in map units is quite exact definition. Also GDAL seems to use the dimension of the square rather than scale, see https://gdal.org/drivers/vector/mvt.html
```
Finnish ETRS-TM35FIN (EPSG:3067) tiling scheme:
{
  "...": "...",
  "crs":"EPSG:3067",
  "tile_origin_upper_left_x":-548576.0,
  "tile_origin_upper_left_y":8388608.0,
  "tile_dimension_zoom_0":2097152.0,
}
```